pkg/kube/client.go

```diff
@@ -194,7 +207,7 @@ func (c *Client) Build(reader io.Reader, validate bool) (ResourceList, error) {
 // occurs, a Result will still be returned with the error, containing all
 // resource updates, creations, and deletions that were attempted. These can be
 // used for cleanup or other logging purposes.
-func (c *Client) Update(original, target ResourceList, force bool) (*Result, error) {
+func (c *Client) Update(original, target ResourceList, force bool, timeout time.Duration) (*Result, error) {
```

Comment:
As you said, this is a breaking change as written. You can work around that by doing something like this:

```suggestion
func (c *Client) Update(original, target ResourceList, force bool) (*Result, error) {
	return UpdateWithTimeout(original, target, force, time.Duration(0))
}	

func (c *Client) UpdateWithTimeout(original, target ResourceList, force bool, timeout time.Duration) (*Result, error) {
```

Where if timeout is 0 you revert to the original behavior.

Reply:
@joejulian thanks for the suggestion. I noticed that there was a similar issue in the meantime which lead to: 
```
type InterfaceExt interface {
	// WaitForDelete wait up to the given timeout for the specified resources to be deleted.
	WaitForDelete(resources ResourceList, timeout time.Duration) error
}
```
So I will follow that pattern and remove the breaking change by introducing a trait like additional interface.

Reply:
@joejulian PR is cleaned up and all breaking changes have been removed as suggested. Looking forward to your feedback.